### PR TITLE
Feature/audio

### PR DESCRIPTION
**Pull request by apoudret-microej**
https://docs.microej.com/en/feature-audio/

### CODE REVIEW

**Review by thomasponsmicroej — APPROVED** [review]

File: ApplicationDeveloperGuide/audio.rst [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+For example, the following snippet records audio with an audio record and writes the audio data to a file:
+
+.. code-block:: java
+
+	try (OutputStream outputStream = new FileOutputStream("record.raw")) {
+	    try (AudioRecord audioRecord = new AudioRecord(FORMAT, 1600)) {
+	        byte[] buffer = new byte[480];
+	        while (true) {
+	            audioRecord.readBuffer(buffer, 0, buffer.length); // read from audio record
+	            outputStream.write(buffer, 0, buffer.length); // write to file
```
Comment (gjadi-mej) [inline ApplicationDeveloperGuide/audio.rst:90]: Write the number of read data only?
```suggestion
	            int bytesRead = audioRecord.readBuffer(buffer, 0, buffer.length); // read from audio record
	            outputStream.write(buffer, 0, bytesRead); // write to file
```

File: VEEPortingGuide/bluetooth.rst [hunk 2]
```diff
@@ -47,7 +39,7 @@ the following dependency to the :ref:`module.ivy <mmm_module_description>` file:
 
 ::
 
-	<dependency org="com.microej.pack.bluetooth" name="bluetooth-pack" rev="2.2.1" />
+	<dependency org="com.microej.pack.bluetooth" name="bluetooth-pack" rev="2.4.1" />
 
 The Platform project must be rebuilt (:ref:`platform_build`).
```
Comment (gjadi-mej) [inline VEEPortingGuide/bluetooth.rst:44]: While you're there, you can remove this line.

**Review by gjadi-mej — CHANGES_REQUESTED** [review]
Nice description of the API.

I've never been a fan of the description of the LLAPI as we don't really document them besides the header files...

File: ApplicationDeveloperGuide/audio.rst (hunk 1 above)
Comment (apoudret-microej) [reply ApplicationDeveloperGuide/audio.rst:90]: readBuffer() blocks until the buffer has been fully filled. the number of bytes returned will be different from buffer size only if the audio record is closed in another thread during the read. in this sample it does not happen, so i think it is not necessary?

File: VEEPortingGuide/bluetooth.rst (hunk 2 above)
Comment (apoudret-microej) [reply VEEPortingGuide/bluetooth.rst]: fixed (amend)

File: ApplicationDeveloperGuide/audio.rst (hunk 1 above)
Comment (gjadi-mej) [reply ApplicationDeveloperGuide/audio.rst:90]: I would say it's better to use the safer pattern just in case code is copied/pasted around :)

Comment (apoudret-microej) [reply ApplicationDeveloperGuide/audio.rst]: that's a good reason.
fixed (amend)

**Review by gjadi-mej — APPROVED** [review]

**Pull request merged**